tests/unit/insights/test_importer.py

```diff
@@ -42,32 +589,239 @@ def get_product_weight_predictions(
+        )
+        get_product_predictions_mock.assert_called_once()
+        import_insights_mock.assert_called_once_with(
+            [], DEFAULT_SERVER_DOMAIN, True, product_store
+        )
+
+    def test_import_insights_single_product(self, mocker):
+        prediction_dict = {
+            "barcode": DEFAULT_BARCODE,
+            "type": PredictionType.category.name,
+            "data": {},
+        }
+        prediction = Prediction(
+            barcode=DEFAULT_BARCODE,
+            type=PredictionType.category,
+            data={},
+        )
+        get_product_predictions_mock = mocker.patch(
+            "robotoff.insights.importer.get_product_predictions",
+            return_value=[
+                prediction_dict,
+            ],
+        )
+        import_insights_mock = mocker.patch(
+            "robotoff.insights.importer.InsightImporter.import_insights",
+            return_value=1,
+        )
+        product_store = FakeProductStore()
+        imported = import_insights_for_products(
+            {DEFAULT_BARCODE: {PredictionType.category}},
+            DEFAULT_SERVER_DOMAIN,
+            automatic=True,
+            product_store=product_store,
+        )
+        assert imported == 1
+        get_product_predictions_mock.assert_called_once()
+        import_insights_mock.assert_called_once_with(
+            [prediction], DEFAULT_SERVER_DOMAIN, True, product_store
+        )
+
+    def test_import_insights_type_mismatch(self, mocker):
+        prediction_dict = {
+            "barcode": DEFAULT_BARCODE,
+            "type": PredictionType.image_orientation.name,
+            "data": {},
+        }
+        get_product_predictions_mock = mocker.patch(
+            "robotoff.insights.importer.get_product_predictions",
+            return_value=[
+                prediction_dict,
+            ],
+        )
+        import_insights_mock = mocker.patch(
+            "robotoff.insights.importer.InsightImporter.import_insights",
+            return_value=0,
+        )
+        product_store = FakeProductStore()
+        imported = import_insights_for_products(
+            {DEFAULT_BARCODE: {PredictionType.image_orientation}},
+            DEFAULT_SERVER_DOMAIN,
+            automatic=True,
+            product_store=product_store,
         )
-        assert len(insights) == 1
-        insight = insights[0]
-        assert insight.automatic_processing is True
-        assert insight.barcode == DEFAULT_BARCODE
-        assert insight.type == "product_weight"
-        assert insight.data == insight_data
-        assert insight.value_tag is None
-        assert insight.reserved_barcode is False
-        assert insight.server_domain == DEFAULT_SERVER_DOMAIN
-        assert insight.server_type == "off"
-        # check that id is a valid UUID
-        uuid.UUID(insight.id)
+        assert imported == 0
+        assert not get_product_predictions_mock.called
+        assert not import_insights_mock.called
```

Comment:
Cool that you added so many tests ! :muscle:

Reply:
Yes finally the `importer.py` file is correctly tested 🎉 
The new architecture really simplifies the testing process :)